mne/time_frequency/tests/test_spectrum.py

```diff
@@ -389,49 +386,107 @@ def test_spectrum_kwarg_triaging(raw):
     raw.plot_psd(ax=axes)
 
 
-def test_spectrum_array_raw(raw_psd):
-    """Test SpectrumArray for Raw-derived spectra."""
-    raw_psd2 = SpectrumArray(
-        data=raw_psd.get_data(),
-        freqs=raw_psd.freqs,
-        info=raw_psd.info,
-    )
-    assert_array_equal(raw_psd.get_data(), raw_psd2.get_data())
-    assert_array_equal(raw_psd.freqs, raw_psd2.freqs)
-
-
-def test_epochs_spectrum_aray(epochs_psd):
-    """Test EpochsSpectrumArray for Epochs-derived spectra."""
-    epochs_psd2 = EpochsSpectrumArray(
-        data=epochs_psd.get_data(),
-        freqs=epochs_psd.freqs,
-        info=epochs_psd.info,
-        events=epochs_psd.events,
-        event_id=epochs_psd.event_id,
-    )
-    assert_array_equal(epochs_psd.get_data(), epochs_psd2.get_data())
-    assert_array_equal(epochs_psd.freqs, epochs_psd2.freqs)
+def _get_psd_varients(raw_psds, epochs_psds):
+    psds = list()
+    for raw_psd in raw_psds:
+        data = raw_psd.get_data()
+        psds.append(  # complex
+            SpectrumArray(
+                data=data,
+                freqs=raw_psd.freqs,
+                info=raw_psd.info,
+                method=raw_psd.method,
+                weights=raw_psd._mt_weights if raw_psd.method == "multitaper" else None,
+            )
+        )
+        psds.append(
+            SpectrumArray(
+                data=((data * data.conj()).real).mean(
+                    axis=1 if raw_psd.method == "multitaper" else 2
+                ),
+                freqs=raw_psd.freqs,
+                info=raw_psd.info,
+                method=raw_psd.method,
+            )
+        )
+    for epochs_psd in epochs_psds:
+        data = epochs_psd.get_data()
+        psds.append(
+            EpochsSpectrumArray(
+                data=data,
+                freqs=epochs_psd.freqs,
+                info=epochs_psd.info,
+                events=epochs_psd.events,
+                event_id=epochs_psd.event_id,
+                method=epochs_psd.method,
+                weights=epochs_psd._mt_weights
+                if epochs_psd.method == "multitaper"
+                else None,
+            )
+        )
+        psds.append(
+            EpochsSpectrumArray(
+                data=((data * data.conj()).real).mean(
+                    axis=2 if epochs_psd.method == "multitaper" else 3
+                ),
+                freqs=epochs_psd.freqs,
+                info=epochs_psd.info,
+                events=epochs_psd.events,
+                event_id=epochs_psd.event_id,
+            )
+        )
+    return psds
 
 
-def test_plot_spectrum_aray(raw_psd, epochs_psd):
-    """Test SpectrumArray and EpochsSpectrumArray plotting."""
-    psds = list()
-    psds.append(
+def test_spectrum_arrays(raw_psds, epochs_psds, tmp_path):
```

Comment:
what is the motivation / rationale for combining these into one monolithic test?  IMO test functions are easier to write and read when they do *less* stuff, not *more*. (asking for reasons here, not necessarily changes)

Reply:
OK. If they really are that similar, the preferred way to deduplicate in tests is to either (1) factor out a helper function, or  (2) use a `@pytest.mark.parametrize` decorator. Running for-loops inside tests is not good because when the test fails it's not obvious which iteration of the loop you were on when it failed, which makes debugging harder.

Reply:
`git grep "def _get_inst"` will show you the way --- `request.getfixturevalue`